include/frozen/bits/basic_types.h

```diff
@@ -143,6 +143,10 @@ class cvector {
 
   template <class... Args>
   constexpr void emplace_back(Args &&...args) {
+    if (dsize_ == N) {
+      FROZEN_THROW_OR_ABORT(std::out_of_range("Emplacing into full vector"));
+    }
+    data_[dsize_].~T();
```

Comment:
I don't think this is right actually. This function populates an unused space, same with `push_back()`. Granted, this class definitely has some gaps for constructor/destructor calls that may need to be fixed, but here you definitely don't want to call the destructor on an object that has not been constructed. That could have bad consequences.

Reply:
That's actually not the case for this object. Because of how the `T data_[N]; ` is declared, the objects are always populated. This isn't like an allocator where you're working on raw data. If you don't call the deconstructor, it will never be called since the placement new operates on raw data.

Reply:
Oh yeah good point. If the class was declared as `uint8_t data_[N * sizeof(T)]` and the did placement-new on each push it would be different, but the way it's implemented the objects are all constructed _before_ they're pushed, so a push operation needs to destroy the previous value. Definitely warrants a comment to explain since this is pretty non-obvious. I guess we don't need to do this in `push_back()` since that's doing an `=` assignment, not a placement new.